daemon/graphdriver/vfs/driver.go

```diff
@@ -37,9 +43,38 @@ func Init(home string, options []string, uidMaps, gidMaps []idtools.IDMap) (grap
 
 	setupDriverQuota(d)
 
+	if !d.quotaSupported() && d.options.quota.Size > 0 {
+		return nil, fmt.Errorf("Storage Option size no supported for vfs: %v", quota.ErrQuotaNotSupported)
+	}
+
 	return graphdriver.NewNaiveDiffDriver(d, uidMaps, gidMaps), nil
 }
 
+func parseOptions(options []string) (*vfsOptions, error) {
+	opts := &vfsOptions{}
+	for _, option := range options {
+		key, val, err := parsers.ParseKeyValueOpt(option)
+		if err != nil {
+			return nil, err
+		}
+		switch key {
+		case "size":
+			size, err := units.RAMInBytes(val)
+			if err != nil {
+				return nil, err
+			}
+			opts.quota.Size = uint64(size)
+		default:
+			return nil, fmt.Errorf("vfs: unknown option %s", key)
```

Comment:
Same as above, wrap in `errdefs.InvalidArgument`

Reply:
Also, I didn't peek at the other errors in this package, but I'm not sure if `vfs` is needed here.